### DISCUSSION TITLE
This plugin needs to handle WP Posts table better - it is deleting woocommerce orders which makes it unusable for most SDLC processes

### DISCUSSION THREAD

**Discussion by jameshrp** [deliciousbrains/wp-migrate-db#133] [Ideas]
"WARNING: The post types you don't select will be absent in the destination posts table after migration."

Can you please update the way you handle wp_posts so that the unselected post types are not completely deleted? This makes this plugin completely unusable in my case, because woocommerce stores order data in posts, and learndash stores group data in posts. You basically have "all or nothing" on post types -> your functionality to only select certain post types is misleading, because you will delete the other post types.

**Comment by kevinwhoffman** [comment]
Hi, I understand you're looking to migrate a database that excludes a post type in its entirety from the source while maintaining existing posts of that type at the destination. I can see why that would be valuable.

The reason this is not possible in WP Migrate DB today is because the plugin does not merge database tables at the destination; it replaces them with temporary tables.

For an explanation of why things work this way, see the section [Why No Merging?](https://deliciousbrains.com/wp-migrate-db-pro/doc/how-it-works/#why-no-merging) from our documentation on [How It Works Exactly](https://deliciousbrains.com/wp-migrate-db-pro/doc/how-it-works/):

> People often ask us if WP Migrate DB Pro can merge two databases. Usually they have a development site that has had some database changes and a live site that has had different changes made to it. This is incredibly difficult and error-prone since each row of data contains a unique ID that is often used to relate that row of data with other rows of data in the database. Attempting to merge two databases would mean that IDs would have to be changed and related data would get lost or mixed up unless WP Migrate DB Pro knew a great deal more about the way your database is set up than it actually does. WP Migrate DB Pro simply transfers data and overwrites any tables in the destination database that have been transferred from the source database.

**Comment by jameshrp** [comment]
Hey Kevin,

Yeah, I see... the merge is difficult and could mess up primary keys with
other databases. It's also just a problem with wordpress plugins in that
the wp_posts table is used for WAY too many things.

However, this completely killed the value of the plugin for me. I wanted to
use the plugin to maintain a staging environment and selectively push
updates to production. But since woocommerce relies on the wp_posts  (which
shouldn't be pushed) and elementor also relies on wp_posts (which SHOULD be
pushed), I can't use this plugin at all. I dedicated a significant amount
of time to try to get it set up right so it was disappointing!

In the end, I've decided to just edit elementor on the live website and use
github + custom plugin process with local development for everything else.
I'm sure a lot of people are in the same situation as me, if you could
figure out the merge it would be hugely valuable.

*James Cribbs | KN4NEH*
*Founder & CEO*
*Ham Radio Prep*
*(e)* ***@***.***
*(a)* 1309 Coffeen Ave Ste 1956
Sheridan, WY 82801
*(p)* 833.938.1656
*(w)* www.HamRadioPrep.com <http://www.hamradioprep.com/>

**Reply by kevinwhoffman** [reply]
Hey James, I appreciate the feedback. Database merging is a challenge we're familiar with and have invested significant time in solving. You might be interested in [this blog post](https://deliciousbrains.com/syncing-wordpress-database-changes-merging/) that recaps our efforts in solving WP database merging over the years, and where we're at today on the subject. It's a problem that is somewhat straightforward to solve when you're just considering the content types of WordPress core or a single plugin, but the complexity balloons when trying to solve this problem in general.
